model/infinispan/src/main/java/org/keycloak/connections/infinispan/TopologyInfo.java

```diff
@@ -63,7 +63,7 @@ public TopologyInfo(EmbeddedCacheManager cacheManager, Config.Scope config, bool
         if (!embedded) {
             Transport transport = cacheManager.getTransport();
             if (transport != null) {
-                nodeName = transport.getAddress().toString();
+                nodeName = cacheManager.getCacheManagerConfiguration().transport().nodeName();
```

Comment:
@mposolda or @hmlnarik can you comment if this is okay?

Reply:
I am not sure from the top of my head. Will be good to try to run server with/without property "-Djboss.node.name=mynode1" for the cluster/non-cluster configuration and check it behaves as expetected. Also check the cluster/cross-dc tests are not broken by this change :)

Reply:
Cluster tests are passing for me locally except for the AuthenticationSessionFailoverClusterTest, which is the one that makes more use of this change.

I'm testing now without this change to see how it goes and try to find another approach to the problem if that is causing issues.

But CI is failing with "failed to update database", which is quite weird as this changeset is not supposed to cause that. Also investigating.